src/mrpro/algorithms/optimizers/pgd.py

```diff
@@ -0,0 +1,158 @@
+"""Proximal Gradient Descent algorithm."""
+
+import math
+from collections.abc import Callable
+
+import torch
+from typing_extensions import Unpack
+
+from mrpro.algorithms.optimizers.OptimizerStatus import OptimizerStatus
+from mrpro.operators import ProximableFunctionalSeparableSum
+from mrpro.operators.Functional import ProximableFunctional
+from mrpro.operators.Operator import Operator
+
+
+class PGDStatus(OptimizerStatus):
+    """Proximal Gradient Descent callback base class."""
+
+    stepsize: float
+    """Current stepsize of the algorithm."""
+
+
+def pgd(
+    f: Operator[torch.Tensor, tuple[torch.Tensor]] | Operator[Unpack[tuple[torch.Tensor, ...]], tuple[torch.Tensor]],
+    g: ProximableFunctional | ProximableFunctionalSeparableSum,
+    initial_value: torch.Tensor | tuple[torch.Tensor, ...],
+    stepsize: float = 1.0,
+    reg_parameter: float = 0.01,
```

Comment:
is it necessary to have the reg parameter here? Is it not possible to have, for example, the following?

```
lambda_reg = 0.1
g = lambda_reg * L1Norm()
```

Reply:
i think it should work as you would expect of you set `g= 0.01*L1Norm()`

Reply:
if `g = lambda_reg * L1Norm()` , then the prox is already done accordingly? If so then I can remove it from the input and then put in the code  
```
x = g_sum.prox(
                *[yi - stepsize * gi for yi, gi in zip(y, gradient, strict=True)], sigma=stepsize
            )
```
as it was the only point where the lambda_reg was used